rcgen/src/lib.rs

```diff
@@ -361,16 +361,18 @@ impl DnType {
 #[derive(Debug, PartialEq, Eq, Hash, Clone)]
 #[non_exhaustive]
 pub enum DnValue {
-	/// A string of characters from the T.61 character set
```

Comment:
the prior order was inspired by the ordering of the `DirectoryString` ASN.1 type in RFC 5280.

Which doesn't list `Ia5String`, but there is a paragraph in the RFC about implementations using using `Ia5String` for emails due to it being able to represent the `@`.

In general, I'm not a fan of reorders like these, as they make review harder and we don't have tools to enforce that the alphabetic order is kept. I'm fine with it this time though.

Reply:
> the prior order was inspired by the ordering of the DirectoryString ASN.1 type in RFC 5280.

Ah I see. In general I would be fine not re-ordering things that are ordered based on an external specification, but in this case since we need a type included that isn't in the list being referenced doing a sort feels like the right call.

> In general, I'm not a fan of reorders like these, as they make review harder

I specifically separated out the reorder into a separate commit to make it easier to review. If you review commit by commit I don't think it's too much of a burden.

> we don't have tools to enforce that the alphabetic order is kept.

True, but we can try to catch it in review. While imperfect, it still feels better than just haphazardly adding to the end each time. 

Since it sounds like you're OK with this instance I'll resolve the feedback and we can revisit case-by-case.